app/src/main/java/com/example/mvvmcleanarchitecture/features/characters/presentation/CharacterViewModel.kt

```diff
@@ -0,0 +1,37 @@
+package com.example.mvvmcleanarchitecture.features.characters.presentation
+
+import androidx.lifecycle.*
+import com.example.mvvmcleanarchitecture.core.base.BaseViewModel
+import com.example.mvvmcleanarchitecture.features.characters.domain.GetCharactersUseCase
+import com.example.mvvmcleanarchitecture.features.characters.domain.model.Character
+import com.example.mvvmcleanarchitecture.features.characters.presentation.model.CharacterDisplayable
+import kotlinx.coroutines.Dispatchers
+
+class CharacterViewModel(private val getCharactersUseCase: GetCharactersUseCase) : BaseViewModel(),
+    DefaultLifecycleObserver {
+
+    private val _characters by lazy {
+        MutableLiveData<List<Character>>().also {
+            getCharacters(it)
+        }
+    }
+
+    val characters: LiveData<List<CharacterDisplayable>> by lazy {
+        _characters.map { characters ->
+            characters.map { CharacterDisplayable(it) }
+        }
+    }
+
+    private fun getCharacters(charactersLiveData: MutableLiveData<List<Character>>) {
+        setPendingState()
+        getCharactersUseCase(
+            params = Unit,
+            scope = viewModelScope,
+            executionDispatcher = Dispatchers.IO
```

Comment:
Jeśli ustawiasz tutaj taki sam argument jaki jest domyślnie, to możesz ten krok pominąć